testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestReadUniontype.java

```diff
@@ -137,6 +139,79 @@ public void testReadUniontype(String storageFormat)
         }
     }
 
+    @Test(dataProvider = "storage_formats", groups = SMOKE)
+    public void testReadUniontypeWithDereference(String storageFormat)
+    {
+        // According to testing results, the Hive INSERT queries here only work in Hive 1.2
+        if (getHiveVersionMajor() != 1 || getHiveVersionMinor() != 2) {
```

Comment:
Wont this fail for version 1.3+ in future?

Reply:
it failed for hive3.